### PR TITLE
client: separate the metrics package

### PR DESCRIPTION
**Pull request by JmPotato**
### What problem does this PR solve?

Issue Number: ref #8690.

### What is changed and how does it work?

```commit-message
Separate the client metrics package.
```

### Check List

Tests

- Unit test
- Integration test

### Release note

```release-note
None.
```

### CODE REVIEW

File: client/metrics/metrics.go [hunk 1]
```diff
@@ -0,0 +1,262 @@
...
+
+	// RequestDurationTSO records the durations of the successful TSO requests.
+	RequestDurationTSO prometheus.Observer
+	// RequestFailedDurationTSO records the durations of the failed TSO requests.
+	RequestFailedDurationTSO prometheus.Observer
+)
+
+func initCmdDurations() {
+	// WithLabelValues is a heavy operation, define variable to avoid call it every time.
+	CmdDurationTSOWait = cmdDuration.WithLabelValues("wait")
```
Comment (okJiang) [inline client/metrics/metrics.go:204]: We can init them in L150-L199 directly

Comment (JmPotato) [reply client/metrics/metrics.go:204]: It is by design that we have a function to do this so we can call `InitAndRegisterMetrics` to reinitialize them with a client option.

**Review by rleungx — APPROVED** [review]

**Review by okJiang — APPROVED** [review]

**Pull request merged**